### PR TITLE
Get predicted ivrobust

### PR DESCRIPTION
**Pull request by strengejacke**

### CODE REVIEW

**Comment by vincentarelbundock** [comment]
Looks great! FYI, the `lm_robust` class from the same package seems to have the same issue.

**Comment by strengejacke** [comment]
There's a slight inconsistency how both function work, however, `lm_robust` now works, too:
``` r
library(estimatr)
library(insight)
library(fabricatr)
set.seed(15)
dat <- fabricate(
  N = 40,
  y = rpois(N, lambda = 4),
  x = rnorm(N),
  z = rbinom(N, 1, prob = 0.4)
)

# Default variance estimator is HC2 robust standard errors
m <- lm_robust(y ~ x + z, data = dat)
model.matrix(m, data = get_datagrid(m, at = "x", include_response = TRUE))
#>    (Intercept)           x    z
#> 1            1 -1.31622439 0.35
#> 2            1 -0.89384629 0.35
#> 3            1 -0.47146819 0.35
#> 4            1 -0.04909009 0.35
#> 5            1  0.37328801 0.35
#> 6            1  0.79566611 0.35
#> 7            1  1.21804421 0.35
#> 8            1  1.64042231 0.35
#> 9            1  2.06280041 0.35
#> 10           1  2.48517851 0.35
#> attr(,"assign")
#> [1] 0 1 2

data(Kmenta, package = "ivreg")

m <- iv_robust(Q ~ P + D | D + F + A,  se_type = "stata", data = Kmenta)
model.matrix(m, data = get_datagrid(m, at = "P", include_response = TRUE))
#> Error in model.matrix.default(m, data = get_datagrid(m, at = "P", include_response = TRUE)): invalid model frame

model.matrix(terms(m), data = get_datagrid(m, at = "P", include_response = TRUE))
#>    (Intercept)         P      D      F    A
#> 1            1  86.49800 97.535 96.625 10.5
#> 2            1  89.49711 97.535 96.625 10.5
#> 3            1  92.49622 97.535 96.625 10.5
#> 4            1  95.49533 97.535 96.625 10.5
#> 5            1  98.49444 97.535 96.625 10.5
#> 6            1 101.49356 97.535 96.625 10.5
#> 7            1 104.49267 97.535 96.625 10.5
#> 8            1 107.49178 97.535 96.625 10.5
#> 9            1 110.49089 97.535 96.625 10.5
#> 10           1 113.49000 97.535 96.625 10.5
#> attr(,"assign")
#> [1] 0 1 2 3 4
```

**Comment by vincentarelbundock** [comment]
wonderful!

File: R/get_predicted_args.R [hunk 1]
```diff
@@ -21,11 +21,12 @@
   }
 
   # ...but not both
-  if (!is.null(dots$type) && !is.null(predict)) {
-    stop(format_message(
-      "`predict` and `type` cannot both be given. The preferred argument for `get_predicted()` is `predict`.",
-      "To use the `type` argument, set `predict = NULL` explicitly, e.g.,:",
-      '`get_predicted(model, predict = NULL, type = "response")`'
+  if (!is.null(dots$type) && !is.null(predict) && isTRUE(verbose)) {
+    warning(format_message(
```
Comment (bwiernik) [inline R/get_predicted_args.R:25]:
```suggestion
    message(format_message(
```

File: tests/testthat/test-get_predicted.R [hunk 2]
```diff
@@ -455,9 +455,9 @@ test_that("lm: get_predicted vs barebones `predict()`", {
 })
 
 
-test_that("using both `predict` and `type` raises an informative error", {
+test_that("using both `predict` and `type` raises an informative warning", {
   mod <- glm(am ~ hp + factor(cyl), family = binomial, data = mtcars)
-  expect_error(get_predicted(mod, predict = "response", type = "response"))
+  expect_warning(get_predicted(mod, predict = "response", type = "response"))
```
Comment (bwiernik) [inline tests/testthat/test-get_predicted.R:460]:
```suggestion
  expect_message(get_predicted(mod, predict = "response", type = "response"))
```

**Review by bwiernik — COMMENTED** [review]
This is a low-key enough issue, that I think a message is more appropriate. Otherwise looks good

**Comment by vincentarelbundock** [comment]
I tried it and it seemed to work well on my computer. Thanks for your work on this!

**Pull request merged**